packages/@lwc/ssr-compiler/src/compile-template/transformers/component/slotted-content.ts

```diff
@@ -77,6 +77,17 @@ const bAddLightContent = esTemplate`
     });
 `<EsCallExpression>;
 
+function getShadowSlottedContent(slottableChildren: IrChildNode[], cxt: TransformerContext) {
+    return optimizeAdjacentYieldStmts(
+        slottableChildren.flatMap((child) => {
+            if (child.type === 'ExternalComponent') {
+                cxt.isSlotted = false;
+            }
```

Comment:
This smells very strange to me – an `ExternalComponent` is essentially just an `Element`, so it should behave the same as them.

Reply:
Here is my attempt to repro an issue with this: https://github.com/salesforce/lwc/pull/5048

Interestingly I could not fix this by adding `|| child.type === 'Element'`. Something else may be at play here.

In general, it also seems odd to me to just set `cxt.isSlotted` to false here.Shouldn't we reset it back when we're done?

Reply:
@nolanlawson `|| child.type === 'Element'` + resetting `cxt.isSlotted` was exactly the fix for the test you introduced in #5048 🥇